### PR TITLE
1 add referrer

### PR DESCRIPTION
**Pull request by jvasquez75**

### CODE REVIEW

File: src/wbreferrer.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    return set(referrer);
+  }
+  //  go get value from storage if its there
+  return fromStorage();
+}
+
+/**
+ * @returns {string}
+ */
+function get() {
```
Comment (gdbrown) [inline src/wbreferrer.js]: eliminate, change "generate" to "get".

File: src/wbreferrer.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  return generate().toString();
+}
+
+export {
+    get,
```
Comment (gdbrown) [inline src/wbreferrer.js]: only expose "get", the rest are internal details that we don't need to expose.

File: src/wbreferrer.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
+/**
+ * @overview
+ * Takes the referring domain(hostname only) to the site and passes the
+ * hostname to googletag/dfp ad calls.
+ *
+ * This script should be included AFTER googletag is loaded.
+ */
+
+let tokenName;
```
Comment (gdbrown) [inline src/wbreferrer.js]: use const and static set to `wbreferrer`.  don't think we need to worry about collision here.

File: src/wbreferrer.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  if (window.sessionStorage) {
+    window.sessionStorage[tokenName] = value;
+  }
+  return value;
+}
+
+/**
+ * @param {string} newTokenName
+ */
+function setTokenName(newTokenName) {
```
Comment (gdbrown) [inline src/wbreferrer.js]: remove, unless you have strong feelings about needing it.

File: tests/wbreferrer.test.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  t.end();
+});
+
+test('wbreferrer - 3rd party site', (t) => {
+  global.window.location = {
+    hostname: 'www.tmz.com',
+  };
+
+  global.document = {
+    referrer: 'http://www.google.com',
```
Comment (gdbrown) [inline tests/wbreferrer.test.js]: can you test with more referrer samples (with/without www.  multiple subdomains and query strings/hash/etc.).

File: src/wbreferrer.js (hunk 1 above)
Comment (gdbrown) [reply src/wbreferrer.js]: @jvasquez75 one last tweak, just `export get` and to use `import wbreferrer from 'wbreferrer';`.  there has no other purpose but to return the referrer if it's in session and isn't the current site.

**Comment by jvasquez75** [comment]
@gdbrown - change have been made

**Pull request merged**